code/src/reducers/todos.js

```diff
@@ -0,0 +1,30 @@
+import { createSlice } from  '@reduxjs/toolkit';
+
+const initialState = {};
+
+export const todos = createSlice({
+  name: 'todos',
+  initialState: initialState, 
+  reducers: {
+    addProject: (state, action) => {
+      const {inputValue} = action.payload;
+      state[inputValue] = [];
+    },
+    addTodo: (state, action) => {
+      const { projectName, itemInfo } = action.payload;
+      state[projectName].push(itemInfo);
+    },
```

Comment:
Try adding a unique id to your todos that you can use instead of the index, eg. a timestamp. In Jennie's codealong from week 15 you can get a hint on how: [https://github.com/Technigo/codealong-week15-20april/blob/master/src/reducers/fridge.js](https://github.com/Technigo/codealong-week15-20april/blob/master/src/reducers/fridge.js)

Also, since you're using local storage I think you need to add the timestamp to your todo's in order to make it change from "a few seconds ago". One idea is to do something like this:
`state[projectName].push({ description: itemInfo.description, done: itemInfo.done, date: itemInfo.date, createdAt: moment() })`
so that it's saved in the state. Maybe that's where you were headed, because I saw that you have this line in TodoItem.js:  `{moment(item.createdAt).fromNow()}`